arcgis-ios-sdk-samples/Analysis/Statistical query (group and sort)/AddStatisticDefinitionsViewController.swift

```diff
@@ -0,0 +1,97 @@
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+protocol AddStatisticDefinitionsViewControllerDelegate: class {
+    func addStatisticDefinitionsViewController(_ addStatisticDefinitionsViewController: AddStatisticDefinitionsViewController, statisticDefinitions: [AGSStatisticDefinition])
+}
+
+class AddStatisticDefinitionsViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+
+    @IBOutlet private var tableView: UITableView!
+    @IBOutlet private weak var fieldNamePicker: HorizontalPicker!
+    @IBOutlet private weak var statisticTypePicker: HorizontalPicker!
+    @IBOutlet private var tableNavigationItem: UINavigationItem!
+    
+    public var fieldNames = [String]()
+    public var statisticDefinitions = [AGSStatisticDefinition]()
+    private var statisticTypes = ["Average", "Count", "Maximum", "Minimum", "StandardDeviation", "Sum", "Variance"]
+
+    // Delegate
+    weak var delegate: AddStatisticDefinitionsViewControllerDelegate?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Setup UI Controls
+        setupUI()
+    }
+    
+    private func setupUI() {
+        //
+        // Add picker options
+        fieldNamePicker.options = fieldNames
+        statisticTypePicker.options =  statisticTypes
+        
+        // Register table cell
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier: "StatisticDefinitionCell")
+    }
+    
+    //MARK: - TableView data source
+    
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return statisticDefinitions.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "StatisticDefinitionCell")!
+        if statisticDefinitions.count > 0 {
+            let statisticDefinition = statisticDefinitions[indexPath.row]
+            let statisticTypeString = statisticTypes[statisticDefinition.statisticType.rawValue]
+            let text = "\(statisticDefinition.onFieldName) (\(statisticTypeString))"
+            cell.textLabel?.text = text
+        }
+        return cell
+    }
+    
+    // MARK: - Actions
+    
+    @IBAction func addStatisticDefinitionAction(_ sender: Any) {
+        //
+        // Add statistic definition
+        let fieldName = fieldNamePicker.options[fieldNamePicker.selectedIndex]
+        let statisticType = AGSStatisticType(rawValue: statisticTypePicker.selectedIndex)
```

Comment:
It would be more "Swifty" to have an `if let` here and perhaps print something to the console if this doesn't construct a valid AGSStatisticType (which would mean the UI had been misconfigured at build time).

Reply:
The `if let` can only be used with `Optional` and it is not the case here.

Reply:
Ah, but it is an optional. You even force-unwrap it on the next line :)

Initializing an enum from a raw value can return nil. `statisticType` is of type `AGSStatisticType?`. The next line would also have to be modified to remove the force unwrap. It'd look like:

``` Swift
if let statisticType = AGSStatisticType(rawValue: statisticTypePicker.selectedIndex) {
    let statisticDefinition = AGSStatisticDefinition(onFieldName: fieldName, statisticType: statisticType, outputAlias: nil)
    statisticDefinitions.append(statisticDefinition)
}
```